src/RA_Defs.cpp

```diff
@@ -6,57 +6,71 @@
 #include <codecvt>
 
 GetParseErrorFunc GetJSONParseErrorStr = GetParseError_En;
+namespace ra {
 
-static_assert(sizeof(BYTE*) == sizeof(char*), "dangerous cast ahead");
-char* DataStreamAsString(DataStream& stream)
-{
-	return reinterpret_cast<char*>(stream.data());
-}
+inline namespace conversions{
+std::string DataStreamAsString(const DataStream& stream) {
 
-std::string Narrow(const wchar_t* wstr)
-{
-	static std::wstring_convert< std::codecvt_utf8_utf16< wchar_t >, wchar_t > converter;
-	return converter.to_bytes(wstr);
+    auto str{ convert_string<std::string>(stream) };
```

Comment:
Make sure these conversions work with UTF-16 surrogates.  many of our rich presence scripts use unicode emojis:
http://docs.retroachievements.org/Rich-Presence/#unicode-standard-symbols

Reply:
utf16 is either wchar_t or char16_t, it will work.

Reply:
I'm not really sure about the narrow/widen anymore, thanks for bringing it up. If I can't figure it I'll put narrow/widen how they were and put more restrictions on convert_string.

Reply:
Those emojis appear now. They won't look exactly how they are on the docs unless you open the text file in a browser or something. But they appear the same using the release version of RA and from my code changes.